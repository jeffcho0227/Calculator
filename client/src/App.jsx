import React from 'react';

class App extends React.Component {
  constructor() {
    super();
  }

  render() {
    console.log('here');
    return(
      <div>
        hello from React
      </div>
    )
  }
}

export default App;