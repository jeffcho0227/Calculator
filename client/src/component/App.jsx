import React from 'react';

class App extends React.Component {
  constructor() {
    super();
  }

  render() {
    console.log('here');
    return(
      <div>
        hello from React 2222
      </div>
    )
  }
}

export default App;