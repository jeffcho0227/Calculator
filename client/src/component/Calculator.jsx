import React from 'react';
import Display from './Display.jsx';

export default class Calculator extends React.Component {
  constructor() {
    super();
    this.state = {
      displayValue: '0', 
      numbers: [],
      operators: [],
      selectedVale: '',
      selectedOp: '',
    };
  }

  render() {
    return(
      <div>
        <Display />
        this is from Calculator
      </div>
    )
  }
}