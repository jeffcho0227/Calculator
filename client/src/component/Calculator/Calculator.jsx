import React from 'react';
import Display from '../Display/Display.jsx';
import Keypad from '../Keypad/Keypad.jsx';
import style from './Calculator.scss';
import Buttons from '../Buttons/Buttons.jsx';

export default class Calculator extends React.Component {
  constructor() {
    super();
    this.state = {
      displayValue: '0', 
      numbers: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0', '.', 'C'],
      operators: ['+', '-', 'x', '/'],
      storedValue: '',
      selectedOp: '',
    };

    this.setOperator = this.setOperator.bind(this);
    this.updateDisplay = this.updateDisplay.bind(this);
    this.callOperator = this.callOperator.bind(this);
  }

  setOperator(value) {
    let { displayValue, selectedOp, storedValue } = this.state;

    if (selectedOp === '') {
      storedValue = displayValue;
      displayValue = '0';
      selectedOp = value;
    } else {
      selectedOp = value;
    }

    this.setState({ displayValue, selectedOp, storedValue }, () => console.log(this.state));
  };

  updateDisplay(val) {
    let { displayValue } = this.state;

    if (val === '.' && displayValue.includes('.')) val = '';

    if (val === 'ce') {
      displayValue = displayValue.substr(0, displayValue.length - 1);
      if (displayValue === '') displayValue = '0';
    } else {
      displayValue === '0' ? (displayValue = val) : (displayValue += val);
    }

    this.setState({ displayValue });
  };

  callOperator() {
    let { displayValue, selectedOp, storedValue } = this.state;
    let updateStoredValue = displayValue;

    displayValue = parseInt(displayValue, 10);
    storedValue = parseInt(storedValue, 10);

    switch(selectedOp) {
      case '+':
        displayValue = storedValue + displayValue;
        break;
      case '-':
        displayValue = storedValue - displayValue;
        break;
      case 'x':
        displayValue = storedValue * displayValue;
        break;
      case '/':
        displayValue = storedValue / displayValue;
        break;
      default: 
        displayValue = '0';
    }

    displayValue = displayValue.toString();
    selectedOp = '';
    if (displayValue === 'NaN' || displayValue === 'Infinity') {
      displayValue = '0';
    }

    this.setState({ displayValue, selectedOp , storedValue: updateStoredValue })
  }

  
  render() {

    const {displayValue, numbers, operators } = this.state;

    return(
      <div>
        <Display displayValue={displayValue}/>
        <Keypad 
          numbers={numbers}
          operators={this.state.operatros}
          setOperator={this.setOperator}
          updateDisplay={this.updateDisplay}
          callOperator={this.callOperator}
        />
      </div>
    )
  }
}