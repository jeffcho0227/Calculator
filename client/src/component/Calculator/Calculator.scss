*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

.btnContainer {
  width: 98%;
  text-align: center; 
  display: flex;
  flex-flow: row;
  justify-content: space-between;
}

.allBtuContainer {
  width: 80%;
  margin: auto;
  text-align: center;
}