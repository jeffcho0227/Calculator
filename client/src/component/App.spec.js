import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from'enzyme';
import App from './App.jsx';
import Calculator from './Calculator/Calculator.jsx';

describe('testing App component', ()=> {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  });

  it('check if the wrapper for App component exist', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('renders one div component', () => {
    expect(wrapper.find('div')).toHaveLength(1);
  });

  it('should render the Calculator Component', () => {
    expect(wrapper.containsMatchingElement(<Calculator />)).toEqual(true);
  });
});