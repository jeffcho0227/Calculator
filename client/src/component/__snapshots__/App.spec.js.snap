// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing App component should render correctly 1`] = `ShallowWrapper {}`;
